### PR TITLE
Item 6474: Update FormulationsTest domain field checks to use fieldNames() helper

### PR DESCRIPTION
**Pull request by cnathe**

### CODE REVIEW

File: test/src/org/labkey/test/tests/idri/FormulationsTest.java [hunk 1]
```diff
@@ -319,16 +320,14 @@ protected void defineParticleSizeAssay()
...
-        assayDesignerPage.goToBatchFields();
-        assertElementPresent(Locator.tagWithClass("div", "domain-form-no-field-panel"));
+        DomainFormPanel batchPanel = assayDesignerPage.goToBatchFields();
+        assertEquals("Batch fields", Collections.emptyList(), batchPanel.fieldNames());
 
         DomainFormPanel runPanel = assayDesignerPage.goToRunFields();
-        assertThat("Missing expected run property", runPanel.getField("IDRIBatchNumber"), CoreMatchers.notNullValue());
+        assertThat("Run fields", runPanel.fieldNames(), hasItems("IDRIBatchNumber"));
 
         DomainFormPanel resultPanel = assayDesignerPage.goToFieldProperties("Result Properties");
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/tests/idri/FormulationsTest.java]:
```suggestion
         DomainFormPanel resultPanel = assayDesignerPage.goToResultFields();
```

File: test/src/org/labkey/test/tests/idri/FormulationsTest.java [hunk 2]
```diff
@@ -388,16 +387,14 @@ protected void defineVisualInspectionAssay()
...
-        assayDesignerPage.goToBatchFields();
-        assertElementPresent(Locator.tagWithClass("div", "domain-form-no-field-panel"));
+        DomainFormPanel batchPanel = assayDesignerPage.goToBatchFields();
+        assertEquals("Batch fields", Collections.emptyList(), batchPanel.fieldNames());
 
         DomainFormPanel runPanel = assayDesignerPage.goToRunFields();
-        assertThat("Missing expected run property", runPanel.getField("LotNumber"), CoreMatchers.notNullValue());
+        assertEquals("Run fields", Arrays.asList("LotNumber"), runPanel.fieldNames());
 
         DomainFormPanel resultPanel = assayDesignerPage.goToFieldProperties("Result Properties");
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/tests/idri/FormulationsTest.java]:
```suggestion
         DomainFormPanel resultPanel = assayDesignerPage.goToResultFields();
```

File: test/src/org/labkey/test/tests/idri/FormulationsTest.java [hunk 3]
```diff
@@ -424,16 +421,14 @@ protected void defineProvisionalHPLCAssay()
...
-        assertElementPresent(Locator.tagWithClass("div", "domain-form-no-field-panel"));
+        DomainFormPanel batchPanel = assayDesignerPage.goToBatchFields();
+        assertEquals("Batch fields", Collections.emptyList(), batchPanel.fieldNames());
 
         DomainFormPanel runPanel = assayDesignerPage.goToRunFields();
-        assertThat("Missing expected run property", runPanel.getField("RunIdentifier"), CoreMatchers.notNullValue());
-        assertThat("Missing expected run property", runPanel.getField("Method"), CoreMatchers.notNullValue());
+        assertThat("Run fields", runPanel.fieldNames(), hasItems("RunIdentifier", "Method"));
 
         DomainFormPanel resultPanel = assayDesignerPage.goToFieldProperties("Result Properties");
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/tests/idri/FormulationsTest.java]:
```suggestion
         DomainFormPanel resultPanel = assayDesignerPage.goToResultFields();
```

File: test/src/org/labkey/test/tests/idri/FormulationsTest.java [hunk 4]
```diff
@@ -581,17 +576,14 @@ protected void defineHPLCAssay()
...
-        assertElementPresent(Locator.tagWithClass("div", "domain-form-no-field-panel"));
+        DomainFormPanel batchPanel = assayDesignerPage.goToBatchFields();
+        assertEquals("Batch fields", Collections.emptyList(), batchPanel.fieldNames());
 
         DomainFormPanel runPanel = assayDesignerPage.goToRunFields();
-        assertThat("Missing expected run property", runPanel.getField("LotNumber"), CoreMatchers.notNullValue());
-        assertThat("Missing expected run property", runPanel.getField("CompoundNumber"), CoreMatchers.notNullValue());
+        assertThat("Run fields", runPanel.fieldNames(), hasItems("LotNumber", "CompoundNumber"));
 
         DomainFormPanel resultPanel = assayDesignerPage.goToFieldProperties("Result Properties");
```
Comment (labkey-tchad) [inline test/src/org/labkey/test/tests/idri/FormulationsTest.java]:
```suggestion
         DomainFormPanel resultPanel = assayDesignerPage.goToResultFields();
```

File: test/src/org/labkey/test/tests/idri/FormulationsTest.java (hunk 1 above)
Comment (cnathe) [reply test/src/org/labkey/test/tests/idri/FormulationsTest.java]: @labkey-tchad 
We had it as using goToResultFields() for some other tests in the previous fb, but there is a subtle difference. The goToResultFields(), which was added for GPAT assay scenarios, uses "Results Properties" and some of these assays have unfortunately decided to drop the "s" and call their third domain "Result Properties". That is why we needed to use the goToFieldProperties() helper instead.

**Pull request merged**